servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SingleConcatWithPublisher.java

```diff
@@ -152,7 +152,12 @@ public void cancel() {
         }
 
         private void emitSingleSuccessToTarget(@Nullable final T result) {
-            target.onNext(result);
+            try {
+                target.onNext(result);
+            } catch (Throwable cause) {
+                target.onError(cause);
+                return;
+            }
             next.subscribeInternal(this);
```

Comment:
If we terminate in case of an error, do we need to subscribe to the `next` publisher? Consider enhancing the test to verify we do not subscribe to the publisher.

Reply:
I missed the `return` statement above but still nice to validate in a test

Reply:
done